datafusion/core/src/bin/print_aggregate_functions_docs.rs

```diff
@@ -0,0 +1,152 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use datafusion::execution::SessionStateDefaults;
+use datafusion_expr::aggregate_doc_sections::doc_sections;
+use datafusion_expr::AggregateUDF;
+use itertools::Itertools;
+use std::fmt::Write as _;
+use std::sync::Arc;
+
+fn main() {
+    let functions = SessionStateDefaults::default_aggregate_functions();
+    let mut docs = "".to_string();
+
+    // doc sections only includes sections that have 'include' == true
+    for doc_section in doc_sections() {
+        // make sure there is a function that is in this doc section
```

Comment:
Perhaps we could change the signature of `documentation()` to return `Option<&Documentation>` so that returning `None` could signal that there was no documentation for the specified function

That would also make it easy to print out the list of functions that don't have documentation (and thus need to be ported here): "INFO: function 'cos' does not have any documentation"

Once the porting is complete, we could change the info to an error which would ensure that all newly added functions were also documented 🤔

Reply:
Signature updated.